notification.go

```diff
@@ -0,0 +1,97 @@
+package ts3
+
+import (
+	"strings"
+)
+
+type notifyEvent string
+
+// Notify event categorys
+const (
+	// ServerEvents registers the following events:
+	// `cliententerview`, `clientleftview`, `serveredited`.
+	ServerEvents notifyEvent = "server"
+
+	// ServerEvents registers the following events:
+	// `cliententerview`, `clientleftview`, `channeldescriptionchanged`, `channelpasswordchanged`
+	// `channelmoved`, `channeledited`, `channelcreated`, `channeldeleted`, `clientmoved`.
+	ChannelEvents notifyEvent = "channel"
+
+	// TextServerEvents registers the `textmessage` event with `targetmode = 3`.
+	TextServerEvents notifyEvent = "textserver"
+
+	// TextChannelEvents registers the `textmessage` event with `targetmode = 2`.
+	//
+	// Notifications are only received for messages that are sent in the channel that the client is in.
+	TextChannelEvents notifyEvent = "textchannel"
+
+	// TextPrivateEvents registers the `textmessage` event with `targetmode = 1`.
+	TextPrivateEvents notifyEvent = "textprivate"
+
+	// TokenUsedEvents registers the `tokenused` event.
+	TokenUsedEvents notifyEvent = "tokenused"
+)
+
+// Notification contains the information of a notify event.
+type Notification struct {
+	Type string
```

Comment:
notifyEvent?

Reply:
No, `notifyEvent` is more like a category of events. I've also documented it/pointed this out in the comments above each const.

I.e: Registering for `event=server` allows you to receive events with the type `cliententerview`, `clientleftview`, `serveredited`.
So `notifyEvent` **is not the same** as `Notification.Type`.

Maybe it would be more intuitive if `notifyEvent` was `notifyCategory` but I've called it this way because the parameter is called event.

Reply:
rgr, notifyCategory definitely makes it clearer.